src/figuresClasses.ts

```diff
@@ -1,11 +1,91 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+function checkLengths(errorMessage: string, ...lengths: number[]): void {
+  if (lengths.some((length) => length <= 0)) {
+    throw new Error(errorMessage);
+  }
+}
+
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    checkLengths('The lengths of the sides must be greater than zero', a, b, c);
+
+    const longestSide = Math.max(a, b, c);
+
+    if (longestSide >= a + b + c - longestSide) {
+      throw new Error(
+        'The longest side of a triangle must be less than a sum of two others',
+      );
+    }
+  }
 
-export class Triangle implements Figure {}
+  getArea(): number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(
+      semiPerimeter *
+        (semiPerimeter - this.a) *
+        (semiPerimeter - this.b) *
+        (semiPerimeter - this.c),
+    );
 
-export class Circle implements Figure {}
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: Shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
```

Comment:
As with the Triangle class, the constructor parameters for 'color' and 'radius' don't need to be manually assigned to class properties since TypeScript's parameter properties feature can be used.